internal/gojs/wasm/wasm.go

```diff
@@ -0,0 +1,526 @@
+}
+
+func NewFuncU32_V(name string, goFunc FuncU32_V, pName string) *wasm.HostFunc {
+	return newFunc(name, (&funcU32_V{f: goFunc, pName: pName}).Call)
+}
+
+type FuncU32_V func(context.Context, api.Module, uint32)
+
+type funcU32_V struct {
+	f     FuncU32_V
+	pName string
+}
+
+// Call implements the same method as defined on api.GoModuleFunction.
+func (f *funcU32_V) Call(ctx context.Context, mod api.Module, wasmStack []uint64) {
+	mem := mod.Memory()
+
+	params := mustReadN(mem, uint32(wasmStack[0]), 1)
+	p := le.Uint32(params)
+	f.f(ctx, mod, p)
+}
+
+func NewFuncV_I64(name string, goFunc FuncV_I64, rName string) *wasm.HostFunc {
+	return newFunc(name, (&funcV_I64{f: goFunc, rName: rName}).Call)
+}
+
+type FuncV_I64 func(context.Context, api.Module) int64
+
+type funcV_I64 struct {
+	f     FuncV_I64
+	rName string
+}
+
+// Call implements the same method as defined on api.GoModuleFunction.
+func (f *funcV_I64) Call(ctx context.Context, mod api.Module, wasmStack []uint64) {
+	mem := mod.Memory()
+
+	stack := mustReadN(mem, uint32(wasmStack[0]), 1)
+	r := f.f(ctx, mod)
+	le.PutUint64(stack, api.EncodeI64(r))
+}
+
+func NewFuncV_I64I32(name string, goFunc FuncV_I64I32, r1Name, r2Name string) *wasm.HostFunc {
+	return newFunc(name, (&funcV_I64I32{f: goFunc, r1Name: r1Name, r2Name: r2Name}).Call)
+}
+
+type FuncV_I64I32 func(context.Context, api.Module) (int64, int32)
+
+type funcV_I64I32 struct {
+	f              FuncV_I64I32
+	r1Name, r2Name string
+}
+
+// Call implements the same method as defined on api.GoModuleFunction.
+func (f *funcV_I64I32) Call(ctx context.Context, mod api.Module, wasmStack []uint64) {
+	mem := mod.Memory()
+
+	stack := mustReadN(mem, uint32(wasmStack[0]), 2)
+	r1, r2 := f.f(ctx, mod)
+	le.PutUint64(stack, api.EncodeI64(r1))
+	le.PutUint64(stack[8:], api.EncodeI32(r2))
+}
+
+func NewRefreshFuncRA_RB(name string, goRefreshFunc RefreshFuncRA_RB, p1Name, p2Name, r1Name, r2Name string) *wasm.HostFunc {
+	fn := &refreshFuncRA_RB{f: goRefreshFunc, p1Name: p1Name, p2Name: p2Name, r1Name: r1Name, r2Name: r2Name}
+	return newFunc(name, fn.Call)
+}
+
+type (
+	RefreshFuncRA_RB func(ctx context.Context, mod api.Module, p1 js.Ref, p2 []js.Ref) (js.Ref, bool)
+	refreshFuncRA_RB struct {
+		f                              RefreshFuncRA_RB
+		p1Name, p2Name, r1Name, r2Name string
+	}
+)
+
+// Call implements the same method as defined on api.GoModuleRefreshFunction.
+func (f *refreshFuncRA_RB) Call(ctx context.Context, mod api.Module, wasmStack []uint64) {
+	mem := mod.Memory()
+
+	paramCount := uint32(3)
+	params := mustReadN(mem, uint32(wasmStack[0]), paramCount)
+	p1 := js.Ref(le.Uint64(params))
+	p2Array := le.Uint32(params[8:])
+	p2Len := le.Uint32(params[16:])
+
+	p2 := refSlice(mem, f.p2Name, p2Array, p2Len)
+
+	r1, r2 := f.f(ctx, mod, p1, p2)
+
+	refreshSPAndStoreResults(mem, mod, paramCount, r1, r2)
+}
+
+func refreshSPAndStoreResults(mem api.Memory, mod api.Module, paramCount uint32, r1 js.Ref, r2 bool) {
```

Comment:
the two functions that use this are gnarly "new" and "call", but now I like that even though the stack game including refreshing the pointer is gnarly, it isn't also mixed in with the business logic, like it was before.